diff --git a/x/emissions/module/abci.go b/x/emissions/module/abci.go
index 94adfd8c..50495035 100644
--- a/x/emissions/module/abci.go
+++ b/x/emissions/module/abci.go
@@ -2,16 +2,22 @@ package module
 
 import (
 	"context"
+	"fmt"
 
 	"cosmossdk.io/errors"
+	emissionskeeper "github.com/allora-network/allora-chain/x/emissions/keeper"
 	allorautils "github.com/allora-network/allora-chain/x/emissions/keeper/actor_utils"
 	"github.com/allora-network/allora-chain/x/emissions/module/rewards"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
 func EndBlocker(ctx context.Context, am AppModule) error {
-
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// Add invariant check logic
+	invariantMessage, invariantFailed := emissionskeeper.AllInvariants(am.keeper)(sdkCtx)
+	if invariantFailed {
+		panic(fmt.Sprintf("Invariants broken: %s", invariantMessage))
+	}
 	blockHeight := sdkCtx.BlockHeight()
 	sdkCtx.Logger().Debug("---------------- Emissions EndBlock -------------------", "blockHeight", blockHeight)
 
