diff --git a/x/emissions/module/abci.go b/x/emissions/module/abci.go
index 80f171e4..7dc2ad15 100644
--- a/x/emissions/module/abci.go
+++ b/x/emissions/module/abci.go
@@ -3,15 +3,23 @@ package module
 import (
 	"context"
 
+	"fmt"
+
 	"cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	emissionskeeper "github.com/allora-network/allora-chain/x/emissions/keeper"
 	allorautils "github.com/allora-network/allora-chain/x/emissions/keeper/actor_utils"
 	"github.com/allora-network/allora-chain/x/emissions/module/rewards"
 )
 
 func EndBlocker(ctx context.Context, am AppModule) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// Add invariant check logic
+	invariantMessage, invariantFailed := emissionskeeper.AllInvariants(am.keeper)(sdkCtx)
+	if invariantFailed {
+		panic(fmt.Sprintf("Invariants broken: %s", invariantMessage))
+	}
 	blockHeight := sdkCtx.BlockHeight()
 	sdkCtx.Logger().Debug("---------------- Emissions EndBlock -------------------", "blockHeight", blockHeight)
 
diff --git a/x/emissions/module/module.go b/x/emissions/module/module.go
index 006eed47..d8533e3b 100644
--- a/x/emissions/module/module.go
+++ b/x/emissions/module/module.go
@@ -44,6 +44,7 @@ var (
 	_ module.HasGenesis       = AppModule{} // nolint: exhaustruct
 	_ appmodule.AppModule     = AppModule{} // nolint: exhaustruct
 	_ appmodule.HasEndBlocker = AppModule{} // nolint: exhaustruct
+	_ module.HasInvariants    = AppModule{} //nolint: exhaustruct
 )
 
 // ConsensusVersion defines the current module consensus version.
@@ -217,3 +218,8 @@ func (am AppModule) EndBlock(ctx context.Context) error {
 	}
 	return err
 }
+
+// RegisterInvariants registers the emissions module invariants.
+func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
+	keeper.RegisterInvariants(ir, &am.keeper)
+}
