diff --git a/x/emissions/module/abci.go b/x/emissions/module/abci.go
index 94adfd8c..50495035 100644
--- a/x/emissions/module/abci.go
+++ b/x/emissions/module/abci.go
@@ -2,8 +2,10 @@ package module
 
 import (
 	"context"
+	"fmt"
 
 	"cosmossdk.io/errors"
+	emissionskeeper "github.com/allora-network/allora-chain/x/emissions/keeper"
 	allorautils "github.com/allora-network/allora-chain/x/emissions/keeper/actor_utils"
 	"github.com/allora-network/allora-chain/x/emissions/module/rewards"
 	emissionstypes "github.com/allora-network/allora-chain/x/emissions/types"
@@ -15,6 +17,11 @@ func EndBlocker(ctx context.Context, am AppModule) error {
 	defer telemetry.ModuleMeasureSince(emissionstypes.ModuleName, telemetry.Now(), telemetry.MetricKeyEndBlocker)
 
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	// Add invariant check logic
+	invariantMessage, invariantFailed := emissionskeeper.AllInvariants(am.keeper)(sdkCtx)
+	if invariantFailed {
+		panic(fmt.Sprintf("Invariants broken: %s", invariantMessage))
+	}
 	blockHeight := sdkCtx.BlockHeight()
 	sdkCtx.Logger().Debug("---------------- Emissions EndBlock -------------------", "blockHeight", blockHeight)
 
diff --git a/x/emissions/module/module.go b/x/emissions/module/module.go
index bbfb75ca..fe6c7bc7 100644
--- a/x/emissions/module/module.go
+++ b/x/emissions/module/module.go
@@ -40,6 +40,7 @@ var (
 	_ module.HasGenesis       = AppModule{} // nolint: exhaustruct
 	_ appmodule.AppModule     = AppModule{} // nolint: exhaustruct
 	_ appmodule.HasEndBlocker = AppModule{} // nolint: exhaustruct
+	_ module.HasInvariants    = AppModule{} //nolint: exhaustruct
 )
 
 // ConsensusVersion defines the current module consensus version.
@@ -190,14 +191,6 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
 
 // EndBlock returns the end blocker for the emissions module.
 func (am AppModule) EndBlock(ctx context.Context) error {
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	defer func() {
-		if r := recover(); r != nil {
-			err := fmt.Errorf("error: %v", r)
-			sdkCtx.Logger().Error("Recover panic in EndBlocker", err)
-		}
-	}()
-
 	err := EndBlocker(ctx, am)
 	if err != nil {
 		sdkCtx := sdk.UnwrapSDKContext(ctx)
@@ -205,3 +198,7 @@ func (am AppModule) EndBlock(ctx context.Context) error {
 	}
 	return err
 }
+
+func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
+	keeper.RegisterInvariants(ir, &am.keeper)
+}
