diff --git a/x/emissions/module/abci.go b/x/emissions/module/abci.go
index 8f743b31..0511e66d 100644
--- a/x/emissions/module/abci.go
+++ b/x/emissions/module/abci.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 
 	"cosmossdk.io/errors"
+	emissionskeeper "github.com/allora-network/allora-chain/x/emissions/keeper"
 	allorautils "github.com/allora-network/allora-chain/x/emissions/keeper/actor_utils"
 	"github.com/allora-network/allora-chain/x/emissions/module/rewards"
 	sdk "github.com/cosmos/cosmos-sdk/types"
@@ -12,6 +13,10 @@ import (
 
 func EndBlocker(ctx context.Context, am AppModule) error {
 	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	invariantMessage, invariantFailed := emissionskeeper.AllInvariants(am.keeper)(sdkCtx)
+	if invariantFailed {
+		panic(fmt.Sprintf("Invariants broken: %s", invariantMessage))
+	}
 	blockHeight := sdkCtx.BlockHeight()
 	sdkCtx.Logger().Debug(
 		fmt.Sprintf("\n ---------------- Emissions EndBlock %d ------------------- \n",
diff --git a/x/emissions/module/module.go b/x/emissions/module/module.go
index 35906b64..16d5dee4 100644
--- a/x/emissions/module/module.go
+++ b/x/emissions/module/module.go
@@ -19,6 +19,15 @@ import (
 	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
 )
 
+var (
+	_ module.HasInvariants = AppModule{}
+)
+
+// RegisterInvariants registers the emissions module invariants.
+func (am AppModule) RegisterInvariants(ir sdk.InvariantRegistry) {
+	keeper.RegisterInvariants(ir, &am.keeper)
+}
+
 var (
 	_ module.AppModuleBasic   = AppModule{}
 	_ module.HasGenesis       = AppModule{}
